### ISSUE TITLE
Storage concurrency not implemented, writes silently failing

### ISSUE THREAD

**Issue by RobinJayaswal** [auth0/auth0-authorization-extension#144]
I have an array of permissions that I am creating programatically through the API. I use a `Promise.map` to loop over each permission and send a request to the API. Each request returns me the permission object and status 200, implying that the operation was successful. However, when I looked at the dashboard it consistently only actually created a subset of the ones it said it did. For example, for an array of six permissions, it would only create between 2 and 5 of them, on at least five different tries (I repeatedly went to the dashboard and deleted all of the permissions and tried again).

I solved this by adding the `concurrency: 1` option to the Promise.map, so that only one request is out at a time. This fixed the issue by leaving more time between the requests (I likely could have put a delay instead), and it now actually creates all the permissions it says it did. Even with concurrency 2 however, it would return success and a permission object, only for it to not actually exist.

Here is part of the code:

```javascript
return Promise.map(permissions, function(permission) {
    
    // if permission exists do nothing
    if (permission exists) {
       return;
    }
    
    return request(permissionOptions)
       .then(function(resp) {
          console.log(resp);
       }
}, {concurrency: 1}) // concurrency must be limited to 1 or only some permissions are created
```

Without concurrency set to 1, for all of my permissions it will log successful creation in the console.log, but not all would exist.

**Comment by sankethkatta** [comment]
Looking through the code, it seems that the issue might be that concurrency checks are bypassed in for the Webtask Storage:

[`server/init.js#L24`](https://github.com/auth0/auth0-authorization-extension/blob/fe560ed22306e374fac31bb0e59c6d64b37bfe39/server/init.js#L24):

```js
    ? new WebtaskStorageContext(storageContext, { force: 1 })
```

The docs on the [Webtask Storage](https://webtask.io/docs/storage) lays out how to resolve conflicts, but the authorization extension seems to just set the `force` flag, rather than dealing with the conflicts.

Is there a reason for bypassing the concurrency checks? This would force clients (as @RobinJayaswal laid out) to serialize all calls to the API.

**Comment by sandrinodimattia** [comment]
Hi @RobinJayaswal @sankethkatta a while ago we fixed this in [auth0-extension-tools](https://github.com/auth0-extensions/auth0-extension-tools/blob/master/README.md#records).

We have now started using the fix here (#153) and released it as 2.4. You can now update it in the gallery
